rust/arrow/src/compute/kernels/take.rs

```diff
@@ -0,0 +1,519 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+//! Defines take kernel for `ArrayRef`
+
+use std::sync::Arc;
+
+use crate::array::*;
+use crate::array_data::ArrayData;
+use crate::buffer::Buffer;
+use crate::builder::*;
+use crate::compute::util::take_index_from_list;
+use crate::datatypes::*;
+use crate::error::{ArrowError, Result};
+
+/// Take elements from `ArrayRef` by supplying an array of indices.
+///
+/// Supports:
+///  * null indices, returning a null value for the index
+///  * checking for overflowing indices
+pub fn take(
+    array: &ArrayRef,
+    index: &UInt32Array,
+    options: Option<&TakeOptions>,
+) -> Result<ArrayRef> {
+    use TimeUnit::*;
```

Comment:
Just a habit, I've become accustomed to importing enums inside the scope that I need them, to avoid repeating `TimeUnit::A`, `TimeUnit::B`, etc. Would you prefer I move it to the top-level?

Reply:
I'm not sure whether there's a "official style guide" on this but it may be better to put it in top-level to 1) make it easier to find out what are the dependencies for a module, 2) potentially avoid repeating the use clause in all the places that need it.